agent/agent.go

```diff
@@ -1510,6 +1510,8 @@ func (a *agent) Collect(ctx context.Context, networkStats map[netlogtype.Connect
 	var mu sync.Mutex
 	status := a.network.Status()
 	durations := []float64{}
+	p2pConns := 0.0
+	derpConns := 0.0
```

Comment:
IMO you should use ints here and cast to float when you write the values to prometheus.